onnxruntime/core/providers/xnnpack/math/gemm.cc

```diff
@@ -0,0 +1,193 @@
+// Copyright(c) Microsoft Corporation.All rights reserved.
+// Licensed under the MIT License.
+
+#include "gemm.h"
+#include "core/framework/transpose_helper.h"
+#include "core/providers/utils.h"
+
+namespace onnxruntime {
+namespace xnnpack {
+
+bool Gemm::IsGemmOnnxNodeSupported(const NodeUnit& node_unit, const GraphViewer& graph) {
+  bool supported = false;
+  const onnxruntime::Node& node = node_unit.GetNode();
+
+  // use do {} while(false) so it's easier to set a breakpoint on the return
+  do {
+    ConstPointerContainer<std::vector<NodeArg*>> input_defs = node.InputDefs();
+
+    const auto alpha = node.GetAttributes().find("alpha");
+    if (!(*alpha).second.has_f() && (*alpha).second.f() != 1.0) break;
+
+    const auto beta = node.GetAttributes().find("beta");
+    if (!(*beta).second.has_f() && (*beta).second.f() != 1.0) break;
+
+    const NodeArg* A_arg = input_defs[0];
+    const NodeArg* B_arg = input_defs[1];
+    const NodeArg* C_arg = input_defs.size() <= 2 ? nullptr : input_defs[2];
+
+    // Right now assuming C matrix is exists
+    if (!C_arg) break;
```

Comment:
why are we exiting if C isn't present? it's optional and a default value can be used (which is 0 as per the onnx spec) if not present. does xnnpack require it to be provided?